index.go

```diff
@@ -25,7 +25,16 @@ func indexHandler(cacheTime time.Duration) (func(w http.ResponseWriter, r *http.
 	if err != nil {
 		return nil, err
 	}
+	if options.cacheTime == 0 {
+		return nil, errors.New("cache time must be set for index handler")
+	}
 	return func(w http.ResponseWriter, r *http.Request) {
-		serveJSONResponse(r.Context(), w, cacheTime, body)
+		// Return error if any query parameter is present
+		if !options.allowUnknownQueryParameters && len(r.URL.Query()) > 0 {
+			badRequest(w, "not supported query parameters")
+			return
+		}
```

Comment:
For endpoints that do not expect any query parameter, they have been updated to fail if there is any query parameter provided in the request.

WDYT?